### PR TITLE
Improve device support and add support for Apple Silicon chipset (`mps`)

### PR DESCRIPTION
**Pull request by gonlairo**
This pull request improves device support for sklearn API models, enabling transfer between CPU, CUDA and Apple Silicon GPU environments.

Fix https://github.com/AdaptiveMotorControlLab/CEBRA-dev/pull/631

### CODE REVIEW

File: cebra/integrations/sklearn/utils.py [hunk 1]
```diff
@@ -114,16 +116,50 @@ def check_device(device: str) -> str:
...
+                    "MPS not available because the current PyTorch install was not "
+                    "built with MPS enabled.")
+            else:
+                raise ValueError(
+                    "MPS not available because the current MacOS version is not 12.3+ "
+                    "and/or you do not have an MPS-enabled device on this machine."
+                )
+
+        return device
+
```
Comment (stes) [inline cebra/integrations/sklearn/utils.py:161]: Can we add tests to catch all the edge cases here? we then need to verify by running on CPU only, CUDA, and MPS chipset.

@gonlairo

**Review by stes — COMMENTED** [review]
Let's add more tests to verify the functionality.

File: cebra/integrations/sklearn/utils.py (hunk 1 above)
Comment (gonlairo) [reply cebra/integrations/sklearn/utils.py:161]: Added more tests @stes. Lmk if this is enough

Comment (gonlairo) [reply cebra/integrations/sklearn/utils.py:161]: Of course to check that all tests pass we need to run them on a device with apple Silicon too.

Comment (stes) [reply cebra/integrations/sklearn/utils.py:161]: looks good

**Review by stes — APPROVED** [review]

**Review by stes — APPROVED** [review]

**Review by MMathisLab — APPROVED** [review]
lgtm - didn't directly test myself though :)

**Comment by gonlairo** [comment]
test with CUDA:
Screen Shot 2023-07-17 at 7 17 29 PM
